### PR TITLE
Configurable Keybindings

### PR DESCRIPTION
**Pull request by ThrosturX**
Almost the same as #610 , except doesn't contain the mode indicator stuff.

### CODE REVIEW

**Comment by jelly** [comment]
This is still all dead code:
```
termite.cc: At global scope:
termite.cc:1029:13: warning: ‘void move_forward_end_blank_word(VteTerminal*, select_info*)’ defined but not used [-Wunused-function]
 static void move_forward_end_blank_word(VteTerminal *vte, select_info *select) {
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~
termite.cc:1025:13: warning: ‘void move_forward_end_word(VteTerminal*, select_info*)’ defined but not used [-Wunused-function]
 static void move_forward_end_word(VteTerminal *vte, select_info *select) {
             ^~~~~~~~~~~~~~~~~~~~~
termite.cc:795:13: warning: ‘long int bottom_row(VteTerminal*)’ defined but not used [-Wunused-function]
 static long bottom_row(VteTerminal *vte) {
             ^~~~~~~~~~
termite.cc:789:13: warning: ‘long int middle_row(VteTerminal*)’ defined but not used [-Wunused-function]
 static long middle_row(VteTerminal *vte) {
             ^~~~~~~~~~
termite.cc:784:13: warning: ‘long int top_row(VteTerminal*)’ defined but not used [-Wunused-function]
 static long top_row(VteTerminal *vte) {
             ^~~~~~~
termite.cc:356:21: warning: ‘const char* binding_get_name(keybinding_cmd)’ defined but not used [-Wunused-function]
 static const char * binding_get_name(keybinding_cmd cmd) {
                     ^~~~~~~~~~~~~~~~
termite.cc: In function ‘void move_to_eol(VteTerminal*, select_info*)’:
termite.cc:2127:1: warning: assuming signed overflow does not occur when changing X +- C1 cmp C2 to X cmp C2 -+ C1 [-Wstrict-overflow]
 }
```

**Comment by jelly** [comment]
These are other compiler warnings caused by this change:
```
                                                                                                                       ^~~~
termite.cc: At global scope:
termite.cc:92:11: warning: no previous declaration for ‘guint vi_mode::parse_mode(const char*)’ [-Wmissing-declarations]
     guint parse_mode ( const char *q )
           ^~~~~~~~~~
termite.cc:129:12: warning: no previous declaration for ‘gchar* vi_mode::to_string(guint)’ [-Wmissing-declarations]
     gchar *to_string ( const guint mode )
            ^~~~~~~~~
termite.cc: In function ‘const char* binding_get_name(keybinding_cmd)’:
termite.cc:359:23: warning: comparison of integer expressions of different signedness: ‘int’ and ‘long unsigned int’ [-Wsign-compare]
     for (int i = 0; i < num_bindings; ++i) {
                     ~~^~~~~~~~~~~~~~
```

**Comment by ThrosturX** [comment]
Hey @jelly , hope you don't mind the mad delays during the summer ;) I don't have access to my workstation right now but I pushed some changes that should address these issues. I'd love to see if there is anything left to complain about, since I'm not in the position to build the project properly right now.

**Comment by ThrosturX** [comment]
Made some updates. README Updated mostly. Seems like all the compiler errors are there without my patch, so ignoring the ones that are left.

**Comment by dsifford** [comment]
@jelly What needs to happen for this PR to merge?

**Comment by ThrosturX** [comment]
Resolved the merge conflict caused by PR #660, this patch already solved that issue so the code was just deleted.

File: termite.cc [hunk 1]
```diff
@@ -809,220 +996,190 @@ gboolean window_state_cb(GtkWindow *, GdkEventWindowState *event, keybind_info *
...
+    if (keybind_exists(bind)) {
+        auto command_details = keybind_get_command(bind);
+        /*
+=======
     if (info->config.fullscreen && event->keyval == GDK_KEY_F11 && !modifiers) {
         info->fullscreen_toggle(info->window);
         return TRUE;
     }
+>>>>>>> thestinger/master
+*/
```
Comment (jelly) [inline termite.cc]: Something is still wrong here :)

File: termite.cc [hunk 2]
```diff
@@ -809,220 +996,190 @@ gboolean window_state_cb(GtkWindow *, GdkEventWindowState *event, keybind_info *
...
+            return TRUE;
     } else if ((modifiers == (GDK_CONTROL_MASK|GDK_MOD1_MASK)) ||
                (modifiers == (GDK_CONTROL_MASK|GDK_MOD1_MASK|GDK_SHIFT_MASK))) {
         if (modify_key_feed(event, info, modify_meta_table))
             return TRUE;
     } else if (modifiers == GDK_CONTROL_MASK) {
+<<<<<<< HEAD
+        if (modify_key_feed(event, info, modify_table))
+            return TRUE;
+=======
```
Comment (jelly) [inline termite.cc]: also small issue here :)

**Review by jelly — CHANGES_REQUESTED** [review]

File: termite.cc (hunk 1 above)
Comment (ThrosturX) [reply termite.cc]: Sorry about that, absolutely my bad.

File: termite.cc (hunk 2 above)
Comment (ThrosturX) [reply termite.cc]: SO sorry about this one, should absolutely have caught that myself, my bad indeed!

**Comment by laktak** [comment]
Would this PR allow us to disable the default keybindings like requested in #507?

Why are the defaults set in code instead of a default config file?

**Comment by codeclem** [comment]
Is this PR missing anything to be merge ready?

**Comment by ThrosturX** [comment]
Do I understand correctly that the requested change is just an update to README.rst?

**Comment by thestinger** [comment]
Termite is obsolete. Please use [Alacritty](https://github.com/alacritty/alacritty) instead. See https://github.com/thestinger/termite#termite-is-obsoleted-by-alacritty for more details.

**Pull request closed without merging**